File: examples/batch-verify.py
"""Verify multiple claims at once."""
from veroq import VeroqClient

client = VeroqClient()

claims = [
    "Tesla delivered over 1.8 million vehicles in 2025",
    "The S&P 500 hit an all-time high in March 2026",
    "Bitcoin crossed $100,000 in 2025",
    "Amazon's AWS revenue exceeds $100 billion annually",
]

print("Batch Claim Verification\n")
print(f"  {'#':<3} {'Verdict':<14} {'Conf':>5}  Claim")
print(f"  {'-'*70}")

for i, claim in enumerate(claims, 1):
    result = client.verify(claim)
    verdict = result.get("verdict", "unknown").upper()
    confidence = result.get("confidence", 0)
    sources = len(result.get("evidence_chain", []))
    short = claim[:50] + ("..." if len(claim) > 50 else "")
    print(f"  {i:<3} {verdict:<14} {confidence*100:>4.0f}%  {short}")
    print(f"      Sources: {sources}")

print(f"\n  Verified {len(claims)} claims.")

File: examples/batch-verify.ts
// Verify multiple claims in parallel
import { VeroqClient } from '@veroq/sdk';

const client = new VeroqClient();

const claims = [
  "NVIDIA is the most valuable company in the world",
  "Bitcoin hit an all-time high in 2026",
  "The Fed raised rates in March 2026",
];

const results = await Promise.all(claims.map((c) => client.verify(c)));

for (let i = 0; i < claims.length; i++) {
  const r = results[i];
  const pct = Math.round(r.confidence * 100);
  console.log(`${r.verdict.padEnd(12)} (${pct}%) — "${claims[i]}"`);
}
